chapter09/mssw/exercise2.pl

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/perl
+
+use v5.12;
+use strict;
+use warnings;
+use diagnostics;
+binmode STDOUT, ':utf8';
+
+use Data::Dumper;
+
+my $patterns_file = 'patterns.dat';
+
+if ( !-f -e $patterns_file ) {
+    print "Unable to find patterns file '$patterns_file'!";
+    exit 1;
+}
+
+print "Compiling patterns...\n";
+
+open my $fh, '<:encoding(UTF-8)', $patterns_file;
+
+my @patterns;
+my $formatting_length = 0;
+while ( my $pattern = <$fh> ) {
+    chomp $pattern;
+    my $regex = qr/$pattern/;
+
+    eval { '' =~ $regex };
+    if ($@) {
+        print STDERR "Invalid pattern '$pattern'\n";
+    }
```

Comment:
Ahh that makes sense, i must have thought it would still need to be tested before being used.